src/query-builder/where-interface.ts

```diff
@@ -344,84 +287,27 @@ export interface WhereInterface<DB, TB extends keyof DB> {
   ): WhereInterface<DB, TB>
 
   /**
-   * Adds a `where exists` clause to the query.
-   *
-   * You can either use a subquery or a raw {@link sql} snippet.
-   *
-   * ### Examples
-   *
-   * The query below selets all persons that own a pet named Catto:
-   *
-   * ```ts
-   * const petName = 'Catto'
-   * const persons = await db
-   *   .selectFrom('person')
-   *   .selectAll()
-   *   .whereExists((qb) => qb
-   *     .selectFrom('pet')
-   *     .select('pet.id')
-   *     .whereRef('person.id', '=', 'pet.owner_id')
-   *     .where('pet.name', '=', petName)
-   *   )
-   *   .execute()
-   * ```
-   *
-   * The generated SQL (PostgreSQL):
-   *
-   * ```sql
-   * select * from "person"
-   * where exists (
-   *   select "pet"."id"
-   *   from "pet"
-   *   where "person"."id" = "pet"."owner_id"
-   *   and "pet"."name" = $1
-   * )
-   * ```
-   *
-   * The same query as in the previous example but with using raw {@link sql}:
-   *
-   * ```ts
-   * import { sql } from 'kysely'
-   *
-   * const petName = 'Catto'
-   * db.selectFrom('person')
-   *   .selectAll()
-   *   .whereExists(
-   *     sql`(select pet.id from pet where person.id = pet.owner_id and pet.name = ${petName})`
-   *   )
-   * ```
-   *
-   * The generated SQL (PostgreSQL):
-   *
-   * ```sql
-   * select * from "person"
-   * where exists (
-   *   select pet.id
-   *   from pet
-   *   where person.id = pet.owner_id
-   *   and pet.name = $1
-   * )
-   * ```
+   * @deprecated Follow [these](https://github.com/koskimas/kysely/releases/tag/0.24.0) instructions to migrate
```

Comment:
Ended up deprecating `whereExists` too. Now we'll only have `where` and `whereRef` (and maybe `andWhere`).